### DISCUSSION TITLE
Unique visitor counting in flutter

### DISCUSSION THREAD

**Discussion by virth** [plausible/analytics#2541] [Anything Else]
Hi everybody

Thank you for creating plausible! Love your product. 

I have a question regarding the unique visitor counting. In your [documentation you explain](https://plausible.io/data-policy#how-we-count-unique-users-without-cookies), that it's done based on the IP adress. In the [official Flutter-Plugin](https://pub.dev/packages/plausible_analytics) it will just send [127.0.0.1 in the x-forwarded-for header](https://github.com/bostrot/flutter_plausible_analytics/blob/1d8a0d88caf0c87f7db74cc8de55a8e79c1d9cb3/lib/plausible_analytics.dart#L52).

Does this still work as intented because some component in the system appends the real external IP to the x-forwarded-for header? Or should we try to figure out the real external IP and send it there?

Thank you in advance
cheers Josh

**Comment by metmarkosaric** [comment]
hi @virth! the Flutter plugin is a community-built plugin so it's not something we have tested ourselves at this stage. you can read more about our events API in our docs here: https://plausible.io/docs/events-api. X-Forwarded-For header needs to have an IP address as otherwise, the visits would look like they're coming from the same visitor for us. could you try and see if the Flutter plugin works as intended and tracks visitors/location correctly please? i cannot see any issues reported in the repo of the plugin so i assume it would work all fine. thanks!
